observable/observable.go

```diff
@@ -69,7 +70,8 @@ func (o Observable) Subscribe(handler rx.EventHandler) <-chan subscription.Subsc
 		}
 
 		// OnDone only gets executed if there's no error.
-		if sub.Error == nil {
+		nilPassed := !reflect.ValueOf(sub.Error).IsValid()
```

Comment:
Why this change? Write test to justify the problem or improvement

Reply:
I’ll add the test for it. Forgot to implement it.

Reply:
Not problem, code review is for that